src/arc.ts

```diff
@@ -101,11 +103,91 @@ export function providerHasConfigUi(provider?: any): any | undefined {
+  if (options && options.showNotification && type)
+    options.showNotification(type, message);
+  else
+    alert(message);
+}
+/**
+ * Function to login to biconomy. For first time users a smart contract wallet will be created.
+ * For the existing users it just returns the user's contract wallet address.
+ * For both cases user will be presented a message to sign on his client wallet.
+ * @param userAccount string User client wallet address.
+ * @param options object Object containing functions for showing notifications.
+ * @returns A promise that resolve to contract wallet address.
+ */
+function biconomyLogin(userAccount: string, options?: IEnableWalletProviderParams) {
+  return new Promise<string>((resolve, reject) => {
+    console.log(`Biconomy login status ${biconomy && biconomy.isLogin}`);
+    if (biconomy && !biconomy.isLogin) {
+      biconomy.getUserContract(userAccount).then((response: any) => {
+        if (!response.userContract) {
+          showUserNotification("Please provide your signature in your wallet to use gas less transactions", options, NotificationStatus.Pending);
+          biconomy.login(userAccount, (error: any, loginResponse: any) => {
+            if (error) {
+              return reject(error);
+            }
+            if (loginResponse && loginResponse.transactionHash) {
+              // First time user. Contract wallet transaction pending.
+              console.log("Your on-chain identity is being created. Please wait.");
+              showUserNotification("Your on-chain identity is being created. Please wait.", options, NotificationStatus.Pending);
+              biconomy.onEvent(biconomy.LOGIN_CONFIRMATION, async () => {
+                // User's Contract Wallet creation successful
+                console.log("Your on-chain identity created successfully");
+                showUserNotification("Your on-chain identity created successfully", options, NotificationStatus.Success);
+                const response = await biconomy.getUserContract(userAccount);
+                if (response.userContract) {
+                  resolve(response.userContract);
+                } else {
+                  reject(response);
+                }
+              });
+            } else if (loginResponse && loginResponse.userContract) {
+              // Existing user login successful
+              showUserNotification("Biconomy login successfull", options, NotificationStatus.Success);
+              resolve(loginResponse.userContract);
+            }
+          });
+
+        } else {
+          resolve(response.userContract);
+        }
+      });
+
+    } else if (biconomy) {
+      biconomy.getUserContract(userAccount).then((response: any) => {
+        console.log(response);
+        if (response.userContract) {
+          resolve(response.userContract);
+        } else {
+          reject(response);
+        }
+      });
+    } else {
+      reject("Biconomy is not initialized");
```

Comment:
what is returned in case of reject?  I wonder if in any case the better pattern might be:

```
return reject("Biconomy is not initialized");
```